gradle/build-logic/src/main/kotlin/app/cash/licensee/GenerateSpdxIdTask.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.licensee
+
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.LIST
+import com.squareup.kotlinpoet.MemberName.Companion.member
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.STRING
+import com.squareup.kotlinpoet.TypeSpec
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+
+@CacheableTask
+abstract class GenerateSpdxIdTask : DefaultTask() {
+  @get:Input
+  abstract val inputJson: Property<String>
+
+  @get:OutputDirectory
+  abstract val generatedSpdx: DirectoryProperty
+
+  init {
+    inputJson.convention(project.resources.text.fromUri("https://spdx.org/licenses/licenses.json").asString())
```

Comment:
Whaaaaaaat is going on here. Gradle has an entire HTTP client hidden in a property factory?

Reply:
I think I actually want to revert to continuing the embed the JSON, except we can embed it into the build-logic for this task to use. I'm generally wary of depending on the internet and would prefer updates be explicit rather than it magically getting new values based on whenever you build.

Reply:
> Whaaaaaaat is going on here. Gradle has an entire HTTP client hidden in a property factory?

Yup, since 2.2.

> I think I actually want to revert to continuing the embed the JSON

So we download the json (manually/ automatically) using Gradle but keep the generator to generate Kotlin code from the Json to have typed Spdx Ids? Sounds okay for me too. 

Yeah, and it can't fail a release due connections problems/new entries.

Reply:
Right. And that's basically the goal. We don't want it to change out from underneath us. Also means old builds are reproducible, although not sure why that would ever be needed.